### PR TITLE
[NPU] fix model llava-onevision-qwen2-7b-ov torch compiles error in npu case

### PR DESCRIPTION
**Pull request by ZeyuanChen2000**
## Motivation

When the model `lmms-lab/llava-onevision-qwen2-7b-ov` is launched in the NPU case, a graph execution error caused by `@torch.compile` occurring in funtion `apply_scaling_penalties`.
image

## Modifications

If `disable=is_npu()` is used, `torch.compile` is completely disabled when the NPU platform is detected, and the `eager` mode is used for execution.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26671256720](https://github.com/sgl-project/sglang/actions/runs/26671256720)
Latest PR Test (Extra): :x: [Run #26671256667](https://github.com/sgl-project/sglang/actions/runs/26671256667)

### CODE REVIEW

File: python/sglang/srt/sampling/penaltylib/repetition_penalty.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import torch
 
 from sglang.srt.sampling.penaltylib.orchestrator import _BatchedPenalizer
-from sglang.srt.utils import get_compiler_backend
+from sglang.srt.utils import get_compiler_backend, is_npu
 
 
-@torch.compile(dynamic=True, backend=get_compiler_backend())
+@torch.compile(dynamic=True, backend=get_compiler_backend(), disable=is_npu())
```
Comment (Hexq0210) [inline python/sglang/srt/sampling/penaltylib/repetition_penalty.py]: Maintain consistent writing style.
@torch.compile(dynamic=True, backend=get_compiler_backend(), disable=_is_npu)
